src/ILLink.Shared/MessageFormat.cs

```diff
@@ -1,10 +1,17 @@
 
 #nullable enable
 
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Resources;
+
 namespace ILLink.Shared
 {
 	internal static class MessageFormat
 	{
+		static readonly ResourceManager sharedStrings = new ResourceManager (Path.GetFileNameWithoutExtension (Assembly.GetExecutingAssembly ().GetManifestResourceNames ().FirstOrDefault (name => name.Contains ("SharedStrings")))!, Assembly.GetExecutingAssembly ());
```

Comment:
This shouldn't be necessary, why can't you call `SharedStrings` directly? Namespace name differences? We can probably resolve this at compile time without reflection.

Reply:
So when is in the context of ILLink.RoslynAnalyzers then the resource being used is ILLink.RoslynAnalyzers.SharedStrings but when the context is Mono.Linker the resource being used is Mono.Linker.SharedStrings
I think if there is a middle layer project like ILLink.Shared and then we take ILLink.Shared.SharedStrings everything would be fine. But without an additional project the only other way I found to get the resource was using reflection

Reply:
You can fix this by changing the csprojs as follows:

```diff
-    <EmbeddedResource Include="../ILLink.Shared/SharedStrings.resx" GenerateSource="true" />
+    <EmbeddedResource Include="../ILLink.Shared/SharedStrings.resx">
+      <GenerateSource>true</GenerateSource>
+      <Namespace>ILLink.Shared</Namespace>
+    </EmbeddedResource>
```